compiler/ballerina-lang/src/main/java/io/ballerina/projects/ConfigReader.java

```diff
@@ -300,11 +300,11 @@ private static Node getVariableNode(int position, Map<Document, SyntaxTree> synt
         for (Map.Entry<Document, SyntaxTree> syntaxTreeEntry : syntaxTreeMap.entrySet()) {
             if (syntaxTreeEntry.getValue().containsModulePart()) {
                 ModulePartNode modulePartNode = syntaxTreeMap.get(syntaxTreeEntry.getKey()).rootNode();
-                List<Node> filteredVarNodes = modulePartNode.members().stream()
+                List<ModuleMemberDeclarationNode> filteredVarNodes = modulePartNode.members().stream()
                         .filter(node -> node.kind() == SyntaxKind.MODULE_VAR_DECL &&
                                 node instanceof ModuleVariableDeclarationNode)
-                        .collect(Collectors.toList());
-                for (Node node : filteredVarNodes) {
+                        .toList();
+                for (ModuleMemberDeclarationNode node : filteredVarNodes) {
                     if (node.location().lineRange().startLine().line() <= position &&
                             node.location().lineRange().endLine().line() >= position) {
                         return node;
```

Comment:
@heshanpadmasiri For me this doesn't seem safe, since for the original version the most inner `if` might not be true for all nodes in the `filteredVarNodes`, and then we would fall through to the next iteration of the outer loop. With the new version, we would directly return `null` if the inner filter returns an empty Stream.
WDYT?